packages/round-manager/src/features/round/ViewApplicationPage.tsx

```diff
@@ -312,6 +343,121 @@ export default function ViewApplicationPage() {
   const registeredGithub =
     application?.project?.projectGithub ?? application?.project?.userGithub;
 
+  const StepStatus: React.FC<{
+    status?: Status;
+    index?: number;
+  }> = ({ status, index }) => {
+    return (
+      <div className="relative">
+        <div
+          className={`flex items-center justify-center rounded-full w-[24px] h-[24px] border-[2px] z-10 relative bg-white
+        ${
+          status === "done" || status === "approved"
+            ? "bg-teal-500 border-teal-500"
+            : status === "current"
+            ? "border-violet-500"
+            : status === "rejected"
+            ? "bg-red-500 border-red-500"
+            : ""
+        }
+        `}
+        >
+          {status === "done" && (
+            <img
+              src={CheckmarkWhite}
+              className="h-3 w-3"
+              alt=""
+              data-testid="status-done"
+            />
+          )}
+          {status === "rejected" && (
+            <img
+              src={Rejected}
+              className="h-3 w-3"
+              alt=""
+              data-testid="status-rejected"
+            />
+          )}
+          {status === "current" && (
+            <div
+              className="rounded-full w-[10px] h-[10px] bg-violet-500"
+              data-testid="status-current"
+            />
+          )}
+          {status === undefined && <div data-testid="status-none" />}
+        </div>
+        {/* Connector */}
+        {index !== 0 && (
+          <div
+            className={`h-[100%] w-[2px] absolute z-5 left-1/2 top-0 ml-[-1px] ${
+              status === "done"
+                ? "bg-teal-500"
+                : status === "rejected"
+                ? "bg-red-500"
+                : "bg-grey-200"
+            }`}
+            style={{
+              transform: "rotate(180deg)",
+              transformOrigin: "top",
+            }}
+          />
+        )}
+      </div>
+    );
+  };
+
+  const Step: React.FC<{
+    title: string;
+    icon: React.ReactNode;
+    text: string | React.ReactNode;
+    status?: Status;
+    index?: number;
+  }> = ({ title, icon, index, text, status }) => {
+    return (
+      <div className={`flex gap-4`} data-testid={`application-step-${title}`}>
+        <StepStatus index={index} status={status} />
+        <div
+          className={`grid gap-3 mt-[5px] pb-[35px] ${
+            !status ? "opacity-[0.6]" : ""
```

Comment:
`status` is a union of string constants and undefined, negating it as if it were a boolean makes it hard to understand the code here.